lib/src/components/list_item/notification_list_item.dart

```diff
@@ -0,0 +1,286 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// List Item for notifications
+class ZetaNotificationListItem extends StatefulWidget {
+  /// Constructor for [ZetaNotificationListItem]
+  const ZetaNotificationListItem({
+    super.key,
+    required this.notificationBadge,
+    required this.message,
+    required this.title,
+    this.notificationTime,
+    this.linkText,
+    this.linkOnClick,
+    required this.buttonOnClick,
+    this.hasMore = false,
+  });
+
+  /// Notification Badge to indicate type of notification or who it's coming from
+  final ZetaNotificationBadge notificationBadge;
+
+  /// Notification message
+  final String message;
+
+  /// Notification title
+  final String title;
+
+  /// Time of notificaiton
+  final String? notificationTime;
+
+  /// Attached link text
+  final String? linkText;
+
+  /// Link onClick method
+  final VoidCallback? linkOnClick;
+
+  /// User actions button on click
+  final VoidCallback buttonOnClick;
+
+  /// If notification is a grouped and there are more notifications
+  final bool? hasMore;
```

Comment:
can we rename this to something like 'showDivider'? Also it doesn't need to be a nullable bool as it's being given a default value